flo-freezer/src/main/java/com/spotify/flo/freezer/PersistingContext.java

```diff
@@ -127,36 +129,29 @@ public static void serialize(Object object, OutputStream outputStream) {
 
   private static Kryo getKryo() {
 
-    // Look up chill-scala using reflection to avoid having a hard dependency on the
-    // chill-scala library and allow this to work with multiple scala versions.
-    Class<?> scalaKryoInstantiatorClass = null;
-    try {
-      scalaKryoInstantiatorClass = Class.forName("com.twitter.chill.ScalaKryoInstantiator");
-    } catch (ClassNotFoundException e) {
-      LOG.debug("Could not find com.twitter.chill.ScalaKryoInstantiator: {}", e.toString());
-    }
-
     final Kryo kryo;
 
     // Instantiate kryo..
-    if (scalaKryoInstantiatorClass!= null) {
+    if (hasChillScala()) {
       LOG.debug("using chill-scala");
       // ...for scala (includes java serializers)
-      final KryoInstantiator instantiator;
-      try {
-        instantiator = KryoInstantiator.class.cast(scalaKryoInstantiatorClass.newInstance());
-      } catch (InstantiationException | IllegalAccessException e) {
-        throw new RuntimeException(e);
-      }
-      kryo = instantiator.newKryo();
+      kryo = ScalaKryoSetup.getKryo();
     } else {
       // ...for java
       LOG.debug("using chill-java");
-      kryo = new Kryo();
+      kryo = new Kryo() {
+        @Override
+        public Registration getRegistration(Class type) {
```

Comment:
It isn't completely valid to override `getRegistration`, because it looks more like default serializer. There is `getDefaultSerializerForAnnotatedType` to override.

Reply:
For instance, this will make not possible to register custom serializers for anything that `JacksonJsonSerializer` can comprehend.

Reply:
Unfortunately Kryo [special handles](https://github.com/EsotericSoftware/kryo/blob/master/src/com/esotericsoftware/kryo/Kryo.java#L486) `Proxy` instances in a way which means that `getDefaultSerializerForAnnotatedType` can't be used to handle `PipelineOptions`.